openfl-tutorials/interactive_api/HPU/PyTorch_TinyImageNet/README.md

```diff
@@ -0,0 +1,140 @@
+- To run the experiment, select the icon that looks like two triangles to "Restart Kernel and Run All Cells". 
+- You will notice activity in your terminals as the experiment runs, and when the experiment is finished the director terminal will display a message that the experiment has finished successfully.  
+
+<br/>
+
+### HPU Tutorials
+#### The name of the file/example that contain HPU adaptations start with "HPU". For example:
+
+ HPU_pytorch_tinyimagenet.ipynb placed under workspace folder of this example is similar to pytorch_tinyimagenet.ipynb but contains HPU adaptations and necessary packages to enable smooth execution on Habana accelerators (Gaudi)
+
+ All the above execution steps remain same for HPU examples but as pre-requisite it needs some additional environment setup and Habana supported package installations which is explained below.
+
+ #### Environment Setup
+
+ To perform an installation of the full driver and SynapseAI software stack independently (as an on-premise installation), you can refer to the dedicated Setup and Install [GitHub repository](https://github.com/HabanaAI/Setup_and_Install) for detailed instructions. Make sure to install the PyTorch and Tensorflow packages provided by Habana. These packages are optimized for Habana Gaudi HPU. Installing public PyTorch and Tensorflow packages is not supported. Refer the same above mentioned github repository for Pytorch and Tensorflow package installations.
+
+ #### HPU Adaptations For Tensorflow Examples
+
+ Call ```load_habana_module()``` function to load the Habana libraries needed to use Gaudi HPU at the TensorFlow level.
+
+ ```
+import tensorflow as tf
+from habana_frameworks.tensorflow import load_habana_module
+load_habana_module()
+ ```
+
+ Once loaded, the Gaudi HPU is registered in TensorFlow and prioritized over CPU. This means that when a given operation is available for both CPU and the Gaudi HPU, the operation is assigned to the Gaudi HPU. Refer [getting started with tensorflow](https://www.intel.com/content/www/us/en/developer/articles/technical/get-started-habana-gaudi-deep-learning-training.html#articleparagraph_cop_1124026931) for detailed explaination and [porting a simple TensorFlow model to Gaudi](https://docs.habana.ai/en/latest/TensorFlow/Migration_Guide/Porting_Simple_TensorFlow_Model_to_Gaudi.html) for sample example
+
+ #### HPU Adaptations For PyTorch Examples
+
+The following set of code additions are required to run a model on Habana. The following steps cover Eager and Lazy modes of execution.
+
+1. Load the Habana PyTorch Plugin Library, libhabana_pytorch_plugin.so.
+
+```
+import torch
+from habana_frameworks.torch.utils.library_loader import load_habana_module
+load_habana_module()
+```
+
+2. Target the Gaudi HPU device:
+
+```
+device = torch.device("hpu")
+```
+3. Move the model to the device:
+
+**This step should be executed before initializing any optimizers. In the upcoming versions the order dependency will be eliminated**
```

Comment:
Habana backend has 2 specific workarounds, as understood from the document.
Suggest continuing the usual steps for PyTorch/Tensorflow, and move the 2 "workarounds" under a separate subsection calling out "Sharp bits" or "Known Issues"
Also worth mentioning if any of them do not apply to Tensorflow (I'm not sure)

Reply:
There is only one workaround to move model to HPU before initializing any optimizers. Like discussed I have mentioned it as workaround in the statement